src/components/search/SearchCourseCard.jsx

```diff
@@ -1,31 +1,143 @@
 import React, { useContext, useMemo } from 'react';
 import PropTypes from 'prop-types';
+import Truncate from 'react-truncate';
 import { Link } from 'react-router-dom';
+import Skeleton from 'react-loading-skeleton';
 import { AppContext } from '@edx/frontend-platform/react';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
 
-const SearchCourseCard = ({ hit }) => {
+import './styles/SearchCourseCard.scss';
+
+const SearchCourseCard = ({ hit, isLoading }) => {
   const { enterpriseConfig: { slug } } = useContext(AppContext);
 
-  const courseDetails = useMemo(
+  const course = useMemo(
     () => {
-      const { title, key } = hit;
-      return { title, key };
+      if (!hit) {
+        return {};
+      }
+      return camelCaseObject(hit);
     },
     [hit],
   );
 
+  const linkToCourse = useMemo(
+    () => {
+      if (!Object.keys(course).length) {
+        return '#';
```

Comment:
make the course link point to nothing while the course is loading (`course` object is empty).